test/database.test.ts

```diff
@@ -175,6 +175,42 @@ describe("database", function () {
       strictEqual(res?.names.length, 1);
     });
 
+    test("when batching single statement, including subselect, affecting two tables throws an error", async function () {
+      // create a "main" (mutable target) table and an "admin" (subselect
+      // target) table
+      const [batchCreate] = await db.batch([
+        db.prepare(`CREATE TABLE main (id INTEGER PRIMARY KEY, data TEXT);`),
+        db.prepare(
+          `CREATE TABLE admin (id INTEGER PRIMARy KEY, address TEXT);`
+        ),
+      ]);
+      const res = await batchCreate.meta.txn?.wait();
+      const names = res?.names ?? [];
+      const tableToMutate = names[0];
+      const tableToSubselect = names[1];
+
+      // try to insert into both the "main" and "admin" tables in a statement
+      const stmt = db.prepare(
+        `insert into ${tableToSubselect} (address) values (?1);insert into ${tableToMutate} (data) select ?2 from ${tableToSubselect} where address=?1;`
+      );
+      const batch = db.batch([stmt.bind(signer.address, "test")]);
+
+      // expect an error due to the individual statements touching two tables
+      await rejects(batch, (err: any) => {
+        strictEqual(
+          err.cause.message,
+          `parsing query: queries are referencing two distinct tables: ${tableToSubselect} ${tableToMutate}`
```

Comment:
noting that this error is thrown instead of `each statement can only touch one table. try batching statements based on the table they mutate.`